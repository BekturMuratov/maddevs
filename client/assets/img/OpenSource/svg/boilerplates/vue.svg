<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>