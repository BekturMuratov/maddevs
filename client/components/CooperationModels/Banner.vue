<template>
  <div class="banner">
    <img
      :data-src="require(`@/assets/img/CooperationModels/jpg/banner.jpg`)"
      class="banner__image img_lazy"
    >
    <div class="container">
      <div class="banner__content">
        <h1 class="banner__title">
          Cooperation <br> models
        </h1>
        <p class="banner__subtitle">
          At Mad Devs, we provide a full range of services aimed at delivering long-term value for both our customers and end users.
        </p>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Banner',
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.banner {
  position: relative;
  min-height: 40vh;
  margin-top: 62px; // height of header component
  padding: 164px 0 139px;
  background: linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #111213 100%);
  overflow: hidden;
  &__image {
    display: block;
    z-index: -1;
    position: absolute;
    top: -2px;
    left: -2px;
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    object-fit: cover;
    object-position: top;
    opacity: 0.48;
  }
  &__content {
    z-index: 2;
    display: block;
    margin: 0 auto;
    max-width: 816px;
    text-align: center;
  }
  &__title {
    color: $text-color--white;
    font-size: 100px;
    line-height: 96px;
  }
  &__subtitle {
    margin-top: 55px;
    font-weight: 600;
    font-size: 32px;
    letter-spacing: -0.013em;
    line-height: 44px;
    color: $text-color--silver;
  }

  @media screen and (max-width: 1024px) {
    &__title {
      font-size: 76px;
      line-height: 76px;
    }
    &__subtitle {
      margin-top: 35px;
      font-size: 24px;
      line-height: 34px;
    }
  }

  @media screen and (max-width: 768px) {
    &__title {
      font-size: 52px;
      line-height: 57px;
    }
    &__subtitle {
      margin-top: 30px;
      font-size: 21px;
      line-height: 29px;
    }
  }
}
</style>
