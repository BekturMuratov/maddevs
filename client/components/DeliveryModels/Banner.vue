<template>
  <div
    id="transparent-header-area"
    class="delivery-model-banner"
  >
    <picture>
      <source
        :srcset="[$getMediaFromS3('/images/DeliveryModels/webp/banner-min.webp') + ' ', $getMediaFromS3('/images/DeliveryModels/webp/banner-min@2x.webp 2x')]"
        type="image/webp"
        class="delivery-model-banner__image"
      >
      <img
        :srcset="$getMediaFromS3('/images/DeliveryModels/jpg/banner-min@2x.jpg')"
        :src="$getMediaFromS3('/images/DeliveryModels/jpg/banner-min.jpg')"
        width="1680"
        height="827"
        class="delivery-model-banner__image"
        alt="Team"
      >
    </picture>
    <div class="container">
      <!-- sectionTextOpacity - value from changeOpacityOnScrollMixin mixin -->
      <div
        ref="sectionText"
        class="delivery-model-banner__content"
        :style="{opacity: sectionTextOpacity}"
      >
        <h1 class="delivery-model-banner__title">
          Delivery <br> models
        </h1>
        <p class="delivery-model-banner__subtitle">
          At Mad Devs, we provide a full range of services aimed at delivering long-term value for both our customers and end users.
        </p>
      </div>
    </div>
  </div>
</template>

<script>
import changeSectionTextOpacityMixin from '@/mixins/changeSectionTextOpacityMixin'

export default {
  name: 'Banner',
  mixins: [changeSectionTextOpacityMixin('sectionText')],
}
</script>

<style lang="scss" scoped>
.delivery-model-banner {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 85vh;
  min-height: 568px;
  padding-top: 62px; // height of header component
  background: linear-gradient(180deg, rgba(17, 18, 19, 0) 60%, #111213 100%);
  overflow: hidden;
  &__image {
    display: block;
    z-index: -1;
    position: absolute;
    top: -2px;
    left: -2px;
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    object-fit: cover;
    object-position: top;
    opacity: 0.48;
  }
  &__content {
    z-index: 2;
    display: block;
    margin: 0 auto;
    max-width: 816px;
    text-align: center;
  }
  &__title {
    color: $text-color--white;
    font-size: 100px;
    line-height: 96px;
    letter-spacing: -0.04em;
  }
  &__subtitle {
    margin-top: 55px;
    font-size: 32px;
    line-height: 44px;
    letter-spacing: -0.013em;
    color: $text-color--silver;
  }

  @media screen and (max-width: 1120px) {
    padding-top: 48px;
  }

  @media screen and (max-width: 1024px) {
    &__title {
      font-size: 80px;
      line-height: 88px;
    }
    &__subtitle {
      margin-top: 35px;
      font-size: 24px;
      line-height: 34px;
    }
  }

  @media screen and (max-width: 768px) {
    &__title {
      font-size: 52px;
      line-height: 57px;
    }
    &__subtitle {
      margin-top: 30px;
      font-size: 21px;
      line-height: 30px;
    }
  }
}
</style>
