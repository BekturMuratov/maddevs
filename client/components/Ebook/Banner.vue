<template>
  <section class="ebook-banner">
    <div class="container">
      <!-- Info -->
      <div class="ebook-banner_info">
        <h1>Custom Software Development: Pricing Strategies</h1>
        <h5>This e-book is for you if you:</h5>
        <ul>
          <li>Are looking for software development services</li>
          <li>Are an aspiring IT company searching for useful information</li>
          <li>Are interested in pricing information for custom software development</li>
        </ul>
        <div class="ebook-banner_info-btns">
          <UIButton>
            Download PDF
            <img
              src="@/assets/img/common/arrow-up-right.svg"
              alt="arrow"
            >
          </UIButton>
          <a href="#">Read it online</a>
        </div>
      </div>

      <!-- Book -->
      <div class="ebook-banner_book">
        <img
          src="@/assets/img/common/book.svg"
          alt="Ebook"
        >
      </div>
    </div>
  </section>
</template>

<script>
import UIButton from '@/components/shared/UIButton'

export default {
  name: 'EbookBanner',
  components: { UIButton },
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

.ebook-banner {
  padding-top: 207px;

  > .container {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
  }

  &_info {
    margin-right: 30px;

    h1 {
      max-width: 650px;
      @include font('Inter', 68px, 600);
      line-height: 72px;
      color: #fff;
      margin-bottom: 72px;
      letter-spacing: -2px;
    }

    h5 {
      @include font('Inter', 20px, 600);
      letter-spacing: -0.4px;
      line-height: 26px;
      color: #fff;
      margin-bottom: 16px;
    }

    ul {
      margin-bottom: 48px;
      list-style: inherit;
      list-style-position: inside;

      li {
        @include font('Inter', 16px, 400);
        line-height: 24px;
        color: #fff;
        letter-spacing: -0.4px;
      }
    }

    &-btns {
      display: flex;
      align-items: center;

      a,
      button {
        @include font('Inter', 16px, 600);
        letter-spacing: -0.4px;
        line-height: 20px;
        color: #fff;
      }

      button {
        display: flex;
        align-items: center;
        padding: 14px 24px;
        margin-right: 32px;

        img {
          margin-left: 7px;
        }
      }

      a {
        text-decoration-line: underline;
        text-underline-offset: 2px;
      }
    }
  }
}
</style>
