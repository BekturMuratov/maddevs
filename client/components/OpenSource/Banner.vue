<template>
  <div
    id="transparent-header-area"
    class="banner"
  >
    <picture>
      <source
        :srcset="[$getMediaFromS3('/images/OpenSource/webp/banner.webp') + ' ', $getMediaFromS3('/images/OpenSource/webp/banner@2x.webp 2x')]"
        type="image/webp"
        class="banner__image"
      >
      <img
        :srcset="$getMediaFromS3('/images/OpenSource/png/banner@2x.png')"
        :src="$getMediaFromS3('/images/OpenSource/png/banner.png')"
        width="1680"
        height="969"
        class="banner__image"
        alt="Programmer"
      >
    </picture>
    <div class="container">
      <!-- sectionTextOpacity - value from changeOpacityOnScrollMixin mixin -->
      <div
        ref="sectionText"
        class="banner__content"
        :style="{opacity: sectionTextOpacity}"
      >
        <h1 class="banner__title">
          The power <br> of open source
        </h1>
        <p class="banner__subtitle">
          70+ of Mad Devs' open source projects have been shared with the community via GitHub.
        </p>
      </div>
    </div>
  </div>
</template>

<script>
import changeSectionTextOpacityMixin from '@/mixins/changeSectionTextOpacityMixin'

export default {
  name: 'Banner',
  mixins: [changeSectionTextOpacityMixin('sectionText')],

  head() {
    return {
      link: [
        {
          rel: 'preload',
          as: 'image',
          href: this.$getMediaFromS3('/images/OpenSource/webp/webp.webp'),
        },

        {
          rel: 'preload',
          as: 'image',
          href: this.$getMediaFromS3('/images/OpenSource/webp/webp@2x.webp'),
        },

        {
          rel: 'preload',
          as: 'image',
          href: this.$getMediaFromS3('/images/OpenSource/png/main.png'),
        },

        {
          rel: 'preload',
          as: 'image',
          href: this.$getMediaFromS3('/images/OpenSource/png/main@2x.png'),
        },
      ],
    }
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/styles/_vars';

.banner {
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-top: 62px;
  height: 100vh;
  min-height: 568px;
  background: linear-gradient(180deg, rgba(17, 18, 19, 0) 75%, #111213 100%);
  overflow: hidden;
  &__image {
    display: block;
    z-index: -1;
    position: absolute;
    top: -2px;
    left: -2px;
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    object-fit: cover;
    object-position: top;
  }
  &__content {
    z-index: 2;
    display: block;
    margin: 0 auto;
    max-width: 816px;
    text-align: center;
  }
  &__title {
    color: $text-color--white;
    font-size: 100px;
    line-height: 96px;
    letter-spacing: -0.04em;
  }
  &__subtitle {
    margin-top: 64px;
    font-size: 32px;
    letter-spacing: -0.013em;
    line-height: 44px;
    color: $text-color--silver;
  }

  @media screen and (max-width: 1024px) {
    height: 85vh;
    &__title {
      font-size: 80px;
      line-height: 88px;
    }
    &__subtitle {
      margin-top: 62px;
      font-size: 24px;
      line-height: 34px;
    }
  }

  @media screen and (max-width: 768px) {
    &__title {
      font-size: 52px;
      line-height: 57px;
    }
    &__subtitle {
      margin-top: 21px;
      font-size: 21px;
      line-height: 30px;
    }
  }
}
</style>
