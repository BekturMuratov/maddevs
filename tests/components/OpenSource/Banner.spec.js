import Banner from '@/components/OpenSource/Banner.vue'
import { render } from '@testing-library/vue'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

describe('Banner component', () => {
  it('should render correctly', () => {
    const { container } = render(Banner, {
      mocks,
      directives: {
        'lazy-load': () => {},
      },
    })

    expect(container).toMatchSnapshot()
  })
})
