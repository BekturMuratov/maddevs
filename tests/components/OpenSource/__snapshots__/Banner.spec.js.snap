// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component should render correctly 1`] = `
<div>
  <div
    class="banner"
    id="transparent-header-area"
  >
    <picture>
      <source
        class="banner__image media_lazy"
        srcset="img.jpg ,img.jpg"
        type="image/webp"
      />
       
      <img
        alt="Developer"
        class="banner__image media_lazy"
        height="969"
        src="img.jpg"
        srcset="img.jpg"
        width="1680"
      />
    </picture>
     
    <div
      class="container"
    >
      <div
        class="banner__content"
        style="opacity: 1;"
      >
        <h1
          class="banner__title"
        >
          
        The power 
          <br />
           of open source
      
        </h1>
         
        <p
          class="banner__subtitle"
        >
          
        70+ of Mad Devs' open source projects have been shared with the community via GitHub.
      
        </p>
      </div>
    </div>
  </div>
</div>
`;
